### PR TITLE
TSQL: INSERT INTO

### LINKED ISSUE #2045: TSQL: INSERT INTO SELECT functionality
https://docs.microsoft.com/en-us/sql/t-sql/statements/insert-transact-sql?view=sql-server-ver15

## Expected Behaviour

The following code parses:
```
BEGIN
  INSERT INTO HumanResources.NewEmployee   
      SELECT EmpID, LastName, FirstName, Phone,   
             Address, City, StateProvince, PostalCode, CurrentFlag  
      FROM EmployeeTemp;  
END
```

## Observed Behaviour
Code does not parse.

### PR DESCRIPTION
**Pull request by jpers36**
Fixes #2045

The ANSI InsertStatementSegment definition uses StartsWith(), which is generally incompatible with the way that we handle TSQL's optional delimitation.  This PR overrides with a TSQL-specific implementation of InsertStatementSegment not using StartsWith(). 

Added a test case proving that this fixes 2045.

Also fixed a trivial typo in ANSI documentation.

### CODE REVIEW

File: test/fixtures/dialects/tsql/insert_statement.sql [hunk 1]
```diff
@@ -28,3 +28,11 @@ SELECT
...
 FROM 
    STAGE.ECDC_CASES
+GO
+
+BEGIN
+  INSERT INTO HumanResources.NewEmployee   
+      SELECT EmpID, LastName, FirstName, Phone,   
+             Address, City, StateProvince, PostalCode, CurrentFlag  
+      FROM EmployeeTemp;  
+END
```
Comment (jpy-git) [inline test/fixtures/dialects/tsql/insert_statement.sql:38]: could we add another test case without the BEGIN END block as well?

```sql
INSERT INTO HumanResources.NewEmployee   
    SELECT
        EmpID,
        LastName,
        FirstName,
        Phone,   
        Address, City,
        StateProvince,
        PostalCode,
        CurrentFlag  
    FROM EmployeeTemp;
```

**Review by jpy-git — COMMENTED** [review]
extra unit test, otherwise looks good 😄

File: test/fixtures/dialects/tsql/insert_statement.sql (hunk 1 above)
Comment (jpers36) [reply test/fixtures/dialects/tsql/insert_statement.sql:38]: Done!

**Review by jpy-git — APPROVED** [review]
Thanks for adding the extra test, the PR is looking good! LGTM! 🥇

**Pull request merged**